src/persist-client/src/cfg.rs

```diff
@@ -296,12 +300,33 @@ impl DynamicConfig {
         self.compaction_minimum_timeout
     }
 
+    /// Fuel added to every compare_and_append call
+    ///
+    /// This is intended to be an analog to `idle_merge_effort`, which in
+    /// differential dataflow configures an arrangement to introduce extra fuel
+    /// on every operator invocation. This causes data in a Trace to continue
+    /// compacting, even when new updates are not being introduced.
+    ///
+    /// Persist doesn't have anything that directly corresponds to operator
+    /// invocations, but we do the best we can to match the spirit of
+    /// idle_merge_effort. Specifically, we introduce the additional fuel on
+    /// each compare_and_append call. In practice, compare_and_append gets
+    /// called on a regular cadence to advance the frontier, so even in no
+    /// updates are being added, we'll eventually compact things down. This does
+    /// mean, however, that tuning the tuning for this constant will likely be
+    /// quite different.
```

Comment:
is there a world in which extra effort is a function of how much uncompacted data we have in the shard? strawman: rather than trying to find a one-size-fits-all constant, we apply proportional effort so that we'd expect any shard, in the absence of new updates, to be fully compacted after N frontier updates (or within M minutes or something)

Reply:
I had the same thought myself, but it would certainly be a divergence from how things work in dd. possible that because we've already (sort of necessarily) diverged, that ship has sailed, but my instinct is to try to be as close as possible

totally down to leave this as a "potentially revisit once we have some concrete experience here" sort of TODO if you like